### PR TITLE
feat(trading): add LP fee settings

### LINKED ISSUE #5552: Support LP fee setting enhancements
## The Chore

Latest Core changes will mean there are now three ways to set fees:
   - Using LP commitments (same as now)
   - Weighted average of the fee values provided on the commitments instead of using the top one
   - Set a single fee on the market manually and update via governance

## Tasks

- [ ] Update the market info to make sure we can see this new market param
- [ ] Add a tooltip offering some explanation of these settings (@candida-d can assist)
- [ ] Check Console 
- [ ] Check Explorer

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #5552 

# Description ℹ️

Add LP fee settings.

# Demo 📺

Screenshot 2024-02-08 at 18 39 34

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: libs/types/src/global-types-mappings.ts [hunk 1]
```diff
@@ -734,3 +735,23 @@ export const ProposalProductTypeShortName: Record<ProposalProductType, string> =
...
+    /** Fee is set by the market to a constant value irrespective of any liquidity provider's nominated fee */
+    METHOD_CONSTANT: 'Constant',
+    /** Fee is smallest value of all bids, such that liquidity providers with nominated fees less than or equal to this value still have sufficient commitment to fulfil the market's target stake. */
+    METHOD_MARGINAL_COST: 'Marginal cost',
+    METHOD_UNSPECIFIED: 'Unspecified',
+    /** Fee is the weighted average of all liquidity providers' nominated fees, weighted by their commitment */
+    METHOD_WEIGHTED_AVERAGE: 'Weighted average',
+  };
+
+export const LiquidityFeeMethodMappingDescription: {
```
Comment (MadalinaRaicu) [inline libs/types/src/global-types-mappings.ts:750]: @candida-d  I have used the descriptions that were generated (coming from the API)
Please feel free to add suggestions if these need changing

**Review by guoguojin — APPROVED** [review]

File: libs/types/src/global-types-mappings.ts [hunk 2]
```diff
@@ -734,3 +735,23 @@ export const ProposalProductTypeShortName: Record<ProposalProductType, string> =
...
+  [e in LiquidityFeeMethod]: string;
+} = {
+  METHOD_CONSTANT: `The fee is set by the market to a constant value irrespective of any liquidity provider's nominated fee.`,
+  METHOD_MARGINAL_COST: `The fee is smallest value of all bids, such that liquidity providers with nominated fees less than or equal to this value still have sufficient commitment to fulfil the market's target stake.`,
+  METHOD_UNSPECIFIED: 'Unspecified',
+  METHOD_WEIGHTED_AVERAGE: `The fee is the weighted average of all liquidity providers' nominated fees, weighted by their commitment`,
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:756]:
```suggestion
  METHOD_CONSTANT: `This liquidity fee is a constant value, set in the market parameters, and overrides the liquidity providers' nominated fees.`,
  METHOD_MARGINAL_COST: `This liquidity fee is the smallest value of all bids from the liquidity providers with enough commitment to fulfil the market's target stake.`,
  METHOD_UNSPECIFIED: 'Unspecified',
  METHOD_WEIGHTED_AVERAGE: `This liquidity fee is the weighted average of all liquidity providers' nominated fees, weighted by their commitment.`,
```

File: libs/types/src/global-types-mappings.ts [hunk 3]
```diff
@@ -734,3 +735,23 @@ export const ProposalProductTypeShortName: Record<ProposalProductType, string> =
...
+
+export const LiquidityFeeMethodMapping: { [e in LiquidityFeeMethod]: string } =
+  {
+    /** Fee is set by the market to a constant value irrespective of any liquidity provider's nominated fee */
+    METHOD_CONSTANT: 'Constant',
+    /** Fee is smallest value of all bids, such that liquidity providers with nominated fees less than or equal to this value still have sufficient commitment to fulfil the market's target stake. */
+    METHOD_MARGINAL_COST: 'Marginal cost',
+    METHOD_UNSPECIFIED: 'Unspecified',
+    /** Fee is the weighted average of all liquidity providers' nominated fees, weighted by their commitment */
+    METHOD_WEIGHTED_AVERAGE: 'Weighted average',
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:747]:
```suggestion
    METHOD_CONSTANT: 'Constant fee',
    /** Fee is smallest value of all bids, such that liquidity providers with nominated fees less than or equal to this value still have sufficient commitment to fulfil the market's target stake. */
    METHOD_MARGINAL_COST: 'Marginal cost fee',
    METHOD_UNSPECIFIED: 'Unspecified',
    /** Fee is the weighted average of all liquidity providers' nominated fees, weighted by their commitment */
    METHOD_WEIGHTED_AVERAGE: 'Weighted average fee',
```

File: libs/types/src/global-types-mappings.ts [hunk 4]
```diff
@@ -734,3 +735,23 @@ export const ProposalProductTypeShortName: Record<ProposalProductType, string> =
...
+  };
+
+export const LiquidityFeeMethodMappingDescription: {
+  [e in LiquidityFeeMethod]: string;
+} = {
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:752]: Could you add a link to the docs here https://docs.vega.xyz/testnet/concepts/liquidity/rewards-penalties#determining-the-liquidity-fee-percentage

Something like "For more detail, visit the docs".

Comment (MadalinaRaicu) [reply libs/types/src/global-types-mappings.ts:752]: added

File: libs/i18n/src/locales/en/trading.json [hunk 5]
```diff
@@ -110,6 +110,7 @@
   "Fills": "Fills",
   "Final commission rate": "Final commission rate",
   "Find out more": "Find out more",
+  "Fore more info, visit the documentation": "For more info, visit the documentation",
```
Comment (candida-d) [inline libs/i18n/src/locales/en/trading.json:113]:
```suggestion
  "For more info, visit the documentation": "For more info, visit the documentation",
```

File: libs/types/src/global-types-mappings.ts [hunk 6]
```diff
@@ -734,3 +735,23 @@ export const ProposalProductTypeShortName: Record<ProposalProductType, string> =
...
+  METHOD_CONSTANT: `This liquidity fee is a constant value, set in the market parameters, and overrides the liquidity providers' nominated fees.`,
+  METHOD_MARGINAL_COST: `This liquidity fee factor is determined by sorting all LP fee bids from lowest to highest, with LPs' commitments tallied up to fulfil the market's target stake. The last LP's bid becomes the fee factor.`,
```
Comment (candida-d) [inline libs/types/src/global-types-mappings.ts:754]:
```suggestion
  METHOD_MARGINAL_COST: `This liquidity fee factor is determined by sorting all LP fee bids with LPs' commitments tallied up to the point of fulfilling the market's target stake. The last LP's bid becomes the fee factor.`,
```

**Review by candida-d — APPROVED** [review]

**Comment by MadalinaRaicu** [comment]
@asiaznik @daro-maj @bwallacee 

Enriches transfer objects with corresponding market and asset objects  #5777  (reload markets and assets for games)

**Review by daro-maj — APPROVED** [review]

**Pull request merged**